sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroidOptions.java

```diff
@@ -170,4 +173,23 @@ public void enableAllAutoBreadcrumbs(boolean enable) {
     enableSystemEventBreadcrumbs = enable;
     enableAppLifecycleBreadcrumbs = enable;
   }
+
+  /**
+   * Returns the Debug image loader
+   *
+   * @return the image loader
+   */
+  public @NotNull IDebugImagesLoader getDebugImagesLoader() {
+    return debugImagesLoader;
+  }
+
+  /**
+   * Sets the image loader
+   *
+   * @param debugImagesLoader the image loader
+   */
+  public void setDebugImagesLoader(final @NotNull IDebugImagesLoader debugImagesLoader) {
+    this.debugImagesLoader =
```

Comment:
`l`: Why not throw an `IllegalArgumentException` when `debugImagesLoader ` is specified as `@NotNull`?

Reply:
we should never `throw` on public APIs, this would be called by the client if they want to.
`@NotNull` is just a helper, compilation won't fail if they really want to pass it null.

https://develop.sentry.dev/sdk/philosophy/#degrade-gracefully